airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -126,22 +126,25 @@ class KubernetesPodOperator(BaseOperator):  # pylint: disable=too-many-instance-
     :type schedulername: str
     :param full_pod_spec: The complete podSpec
     :type full_pod_spec: kubernetes.client.models.V1Pod
-    :param do_xcom_push: If True, the content of the file
-        /airflow/xcom/return.json in the container will also be pushed to an
-        XCom when the container completes.
-    :type do_xcom_push: bool
     :param init_containers: init container for the launched Pod
     :type init_containers: list[kubernetes.client.models.V1Container]
     :param log_events_on_failure: Log the pod's events if a failure occurs
     :type log_events_on_failure: bool
+    :param do_xcom_push: If True, the content of the file
+        /airflow/xcom/return.json in the container will also be pushed to an
+        XCom when the container completes.
+    :type do_xcom_push: bool
+    :param pod_template_file: path to pod template file
+    :type pod_template_file: str
     """
-    template_fields = ('cmds', 'arguments', 'env_vars', 'config_file')
+    template_fields = ('cmds', 'arguments', 'env_vars', 'config_file', 'pod_template_file')
+    template_ext = ('.yaml', '.yml')
```

Comment:
Is templating YAML files such a great idea? It means we have to start worrying about generating valid yaml from templates etc, and that we don't use `{{` elsewhere etc.

Reply:
I guess it is called pod_template_file so this isn't a surprise, but can we please make this clearer in the docs (that this file will be templated.........)

Does making this a template change how Airflow handles it. Rather than this attribute containing a string of the file path Airflow will make it contain the result of rendering the template. That will break the rest of the code won't it?

Reply:
I don't think this would be a problem (though it gives some more flexibility to the users that they need to be careful with). I could see a situation where based on certain settings I might want to use a different pod configuration.

Reply:
```suggestion
```

As I think this will break it. Once this has gone feel free to merge away.

Sorry this has taken me _so many times_ to actually get around to looking at your changes, and thanks for the paitence @davlum